src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -718,6 +778,113 @@ private static bool IsW3CId(string id)
                    ('0' <= id[1] && id[1] <= '9' || 'a' <= id[1] && id[1] <= 'e');
         }
 
+        /// <summary>
+        /// Dispose will stop the Activity if it is already started and notify any event listeners. Nothing will happen otherwise.
+        /// </summary>
+        public void Dispose()
+        {
+            if (!IsFinished)
+            {
+                Stop();
+                if (_source != null)
+                {
+                    _source.RaiseActivityEvent(this, ActivitySourceEventArgs.s_activityStopped);
```

Comment:
Nit:
?.

Reply:
good idea. I had some code in this if block which I removed it, that is why I didn't use ? operator. But now it make sense to use it.

---
In reply to: [383014465](https://github.com/dotnet/runtime/pull/32660#discussion_r383014465) [](ancestors = 383014465)